File: EPI/EvenOdd.py
def evenOdd(arr):
	evenIdx, oddIdx = 0, len(arr) - 1

	while evenIdx < oddIdx:
		if evenIdx % 2 == 0:
			evenIdx += 1
		else:
			arr[evenIdx], arr[oddIdx] = arr[oddIdx], arr[evenIdx]
			oddIdx -= 1

File: EPI/EvenOdd.cpp
#include <vector>
#include <iostream>

using namespace std;

void EvenOdd(vector<int> &vec) {
	int evenIdx = 0, oddIdx = vec.size() - 1;

	while (evenIdx < oddIdx) {
		if (vec[evenIdx] % 2 == 0) {
			evenIdx++;
		} else {
			swap(vec[evenIdx], vec[oddIdx]);
			oddIdx--;
		}
	}
}

int main() {
	vector<int> v {1, 2, 3, 4, 5, 6, 7, 8};
	EvenOdd(v);

	for (int i : v) {
		cout << i << endl;
	}
}